pkg/services/contexthandler/contexthandler.go

```diff
@@ -540,18 +547,25 @@ func (h *ContextHandler) initContextWithToken(reqContext *contextmodel.ReqContex
 
 func (h *ContextHandler) deleteInvalidCookieEndOfRequestFunc(reqContext *contextmodel.ReqContext) web.BeforeFunc {
 	return func(w web.ResponseWriter) {
+		if h.features.IsEnabled(featuremgmt.FlagClientTokenRotation) {
+			return
+		}
+
 		if w.Written() {
 			reqContext.Logger.Debug("Response written, skipping invalid cookie delete")
 			return
 		}
 
 		reqContext.Logger.Debug("Expiring invalid cookie")
-		cookies.DeleteCookie(reqContext.Resp, h.Cfg.LoginCookieName, nil)
+		authn.DeleteSessionCookie(reqContext.Resp, h.Cfg)
 	}
 }
 
 func (h *ContextHandler) rotateEndOfRequestFunc(reqContext *contextmodel.ReqContext) web.BeforeFunc {
 	return func(w web.ResponseWriter) {
+		if h.features.IsEnabled(featuremgmt.FlagClientTokenRotation) {
```

Comment:
To be sure I have correctly understood: this is the condition that makes sure we can only have a Set-cookie returned from the 2 endpoints specific to ClientTokenRotation ?

Reply:
Yes in combination with https://github.com/grafana/grafana/blob/kalleep/auth-token/rotate/pkg/services/contexthandler/contexthandler.go#L495-L500. That one checks if the token needs rotation and short circut if we do.

Reply:
Thank you